### ISSUE TITLE
Account option "Hide your profile details from anonymous viewers?" removes posts from local community

### ISSUE THREAD

**Issue by guzzisti** [friendica/friendica#7852]
### Expected behavior
When the option "Hide your profile details from anonymous viewers?" is set to 'yes', public posts are still shown on the "local community" page `/community/local`.

### Actual behavior
When the option "Hide your profile details from anonymous viewers?" is set to 'yes', public posts are filtered in the view `/community/local`.
This contradictionary to the option description provided in the settings page:

> Anonymous visitors will only see your profile picture, your display name and the nickname you are using on your profile page. Your public posts and replies will still be accessible by other means.

### additional info
Forum Thread: https://sozialwolke.net/display/703b8751-185d-bb47-e0e6-41a769384010

The setting mentioned above set `user.hidewall` to *1*, posts are filtered from the community view because of https://github.com/friendica/friendica/blob/11e71780b97811af7a3042dd3fda9119b7d4e149/mod/community.php#L224

### Friendica version you encountered the problem
2019.12-dev

### Friendica source (git, zip)
GIT

### PHP version
7.3

### SQL version
MySQL 5.7

**Comment by annando** [comment]
We should have a look what hidewall is doing at the different places. Then we should judge, if it makes sense there.

**Comment by annando** [comment]
The background for the current behaviour is that the hiding of the profile mostly is meant to hide the public posts of the selected profile for visitors. But especially on smaller servers this could easily be circumvented by going to the local community page.

I would like to remove this functionality completely. It doesn't serve any real purpose, just some "privacy simulation", nothing more.

**Comment by MrPetovan** [comment]
We still should provide a real "protected account" feature where former posts, even public, aren't accessible on the node anymore. It is a popular feature on Twitter and Mastodon when someone is suddenly a target of harassment. The measure can be temporary so the ACL shouldn't be used for this.

**Comment by annando** [comment]
How can this "protection" work on distributed systems? The posts already left the system, so anyone can comment on the public posts. Where is the corresponding setting on Mastodon? Then I could play with this.

**Comment by MrPetovan** [comment]
That's why I said `accessible on the node`. I'm well aware about the lack of control over data that already left the node.

> Where is the corresponding setting on Mastodon? Then I could play with this.

Ah, never mind, it's only on Twitter. The related settings for Mastodon are:
- Lock account: Requires you to manually approve followers.
- Hide your network: Who you follow and who follows you will not be shown on your profile.
- List this account on the directory: The profile directory is another way by which your account can reach a wider audience.
- Opt-out of search engine indexing: Affects your public profile and status pages.

A eventual "Protected account mode" would temporarily:
- Lock your account
- Hide your network
- Unlist the account from the directories (both local and global)
- Opt-out of search engine indexing

And:
- Hide profile details.
- Hide all posts (including public) from non-followers on the "Status" page and local community.
- Prevent anonymous access to every single post including public) on the node.

Maybe we can propagate the setting to other Friendica nodes to prevent access to distributed public posts but that's pretty much the extent we can do.

**Comment by annando** [comment]
We cannot prevent anonymous access to single public posts. The only option would result in a freezing of this account so it couldn't communicate anymore. With the hiding of the public posts from the community page we are reaching the point of this issue again. There are different expectations about the "hide wall" setting. And the hiding of profile details is just some placebo as well, since basic data need to be shared so that the account still works. Not sharing other data like the name or the avatar would result in a loss of this information even for your followers.

And concerning the 4 settings: We already do have them.

**Comment by MrPetovan** [comment]
> We cannot prevent anonymous access to single public posts.

Yes we do, we just throw an Exception in `mod/display` if the author's user account is "protected".

> The only option would result in a freezing of this account so it couldn't communicate anymore.

Except direct distribution I suppose? But yeah, that's pretty much the goal.

> With the hiding of the public posts from the community page we are reaching the point of this issue again. There are different expectations about the "hide wall" setting.

Yes, because the current setting has a misleading name and incomplete explanation of the effects. But it has nothing to do with my suggestion.

> And the hiding of profile details is just some placebo as well, since basic data need to be shared so that the account still works. Not sharing other data like the name or the avatar would result in a loss of this information even for your followers.

I'm not talking about Display Name or Avatar, just everything beyond that that will be in the upcoming custom profile field feature.

**Comment by annando** [comment]
With AP public posts should be fetchable anonymously via AP. Blocking this could result in federation problems with other systems, possibly even the direct (public) wouldn't work correctly anymore with all receivers. It is possible that some systems on the other side are fetching the content from your site to validate them.

When some account would had this new setting activated, it had to disallow public posts at all. This is the only option. And we had to cut every possibility to fetch content, even from the technical endpoints like the feed or the AP endpoints.

**Comment by MrPetovan** [comment]
Yes, I realize it is a drastic setting, albeit it should be a temporary one, and it should be explained as such.

**Comment by annando** [comment]
Worst thing about this is: This setting won't prevent your existing posts from being viewable on every other system.

But: This whole stuff is not in the scope of this issue here.

**Comment by MrPetovan** [comment]
Yes, again the setting would have to be clearly described, which is kind of the issue with the `hidewall` setting as well which I think we should remove as is.

**Comment by annando** [comment]
I suggest that we create a new issue where we describe the effect of the corresponding settings and how exactly they should work.

**Comment by MrPetovan** [comment]
Good idea.
